### PR TITLE
Migrate to Micronaut

### PR DESCRIPTION
**Pull request by Brutus5000**

### CODE REVIEW

File: src/main/kotlin/com/faforever/userservice/security/FafScope.kt [hunk 1]
```diff
@@ -1,10 +1,7 @@
 package com.faforever.userservice.security
 
-import com.faforever.userservice.security.OAuthScope.SCOPE_PREFIX
-import org.springframework.security.core.GrantedAuthority
-
-data class FafScope(val role: String) : GrantedAuthority {
-    override fun getAuthority() = SCOPE_PREFIX + role
+data class FafScope(val role: String) {
+//    override fun getAuthority() = SCOPE_PREFIX + role
```
[the remark this answers is not in the document]
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/userservice/security/FafScope.kt:4]: The whole class might be obsolete now. I will check

File: src/main/kotlin/com/faforever/userservice/web/OAuthController.kt [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    @Post("/login", consumes = [MediaType.APPLICATION_FORM_URLENCODED])
+    @PermitAll
+    fun performLogin(
+        @Body loginForm: LoginForm,
+        @Header("X-Real-Ip") reverseProxyIp: String?,
+        @QueryValue("_csrf") csrfToken: String,
+        request: HttpRequest<Any>,
+    ): Mono<HttpResponseWithModelView> {
+        val ip = if (reverseProxyIp != null) reverseProxyIp else {
```
[the remark this answers is not in the document]
Comment (Brutus5000) [reply src/main/kotlin/com/faforever/userservice/web/OAuthController.kt:81]: You'd have to use weird other syntax to get the log.warn though (e.g. use an `.also {}` on the fallback value) . Classic if/else seems more straighforward

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -1,36 +1,58 @@
...
 
 faf:
-  passwordResetUrl:  ${PASSWORD_RESET_URL:https://faforever.com/account/password/reset}
-  registerAccountUrl: ${REGISTER_ACCOUNT_URL:https://faforever.com/account/register}
-  accountLinkUrl: ${ACCOUNT_LINK_URL:https://www.faforever.com/account/link}
-
+  passwordResetUrl:  ${PASSWORD_RESET_URL:`https://faforever.com/account/password/reset`}
+  registerAccountUrl: ${REGISTER_ACCOUNT_URL:`https://faforever.com/account/register`}
+  accountLinkUrl: ${ACCOUNT_LINK_URL:`https://www.faforever.com/account/link`}
+  hydra-base-url: http://localhost:4445
```
[the remark this answers is not in the document]
Comment (Brutus5000) [reply src/main/resources/application.yml:49]: Needs to be fixed indeed

**Review by bukajsytlos — APPROVED** [review]
lgtm

**Pull request merged**